llvm/lib/Target/AIE/AIEAddressSpaceLowering.cpp

```diff
@@ -0,0 +1,122 @@
+//===--------- AIEAddressSpaceLowering.cpp - Retype pointer to P0  --------===//
```

Comment:
Not really important, but maybe "Lowering" is a too heavy term that give us a impression that operations will be broken into smaller ones. What happens here is retype or "address space flattening".

Reply:
I proposed this new name back then, but I agree something like `AddressSpaceFlattening` or `AddressSpaceRetyping` sounds better. I would be fine to keep the name like this though, as there is a comment at the top of the file that briefly explains what the pass does.

Reply:
have renamed to `AddressSpaceFlattening`